### DISCUSSION TITLE
Namespace inheritance by children element

### DISCUSSION THREAD

**Discussion by m-valeev** [eclipse-ee4j/jaxb-ri#1778] [Q&A]
For now in version jaxb-runtime-4.0.3 nested elements don't inherit the root element namespace.

```
@XmlAccessorType(XmlAccessType.FIELD)
@XmlType(name = "", propOrder = {

})
@XmlRootElement(name = "ResponseType", namespace = "http://www.anyurl.com/wsdl/")
public class ResponseType {

    @XmlElement(required = true)
    protected String status;
    @XmlElement(required = true)
    protected String message;
```
Status and message will get "" namespace instead of XmlRootElement's namespace so i get unmarshalleing error if try to unmarshall:
```
<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:wsdl="http://www.anyurl.com/wsdl/">
    <soapenv:Header/>
    <soapenv:Body>
        <wsdl:ResponseType>
            <wsdl:status>01</wsdl:status>
            <wsdl:message>success</wsdl:message>
        </wsdl:ResponseType>
    </soapenv:Body>
</soapenv:Envelope>
```
`Unmarshalling Error: unexpected element (uri:"http://www.anyurl.com/wsdl/", local:"status"). Expected elements are <{}message>,<{}status> `

Is it ok?

**Comment by lukasj** [comment]
use 4.0.4 which has been released last week

**Reply by m-valeev** [reply]
sorry i meant that the trouble is in 4.0.4 version.

**Comment by m-valeev** [comment]
@lukasj sorry. I meant i have the problem with 4.0.4 version.

**Reply by m-valeev** [reply]
it is because in my class i don't specify namespace on field explicitly then `uri = ""` is used in `calcXmlName`.
https://github.com/eclipse-ee4j/jaxb-ri/commit/2269a1d80914221a1be73026b7c2e94bb0a4f11f

**Reply by m-valeev** [reply]
is it correct behaviour?
@lukasj

**Reply by laurentschoelens** [reply]
I'm pretty sure 4.0.3 is faulty version and correct behavior (correct = expected since like that from a long time here) is in 4.0.2 and 4.0.4

See https://github.com/eclipse-ee4j/jaxb-ri/pull/1752 which is related to your problem
